src/dp_flow.c

```diff
@@ -133,6 +135,7 @@ static __rte_always_inline void dp_mark_vnf_type(struct dp_flow *df, const struc
 		else
 			key->vnf_type = DP_VNF_TYPE_UNDEFINED;
 	} else if (key->l3_type == RTE_ETHER_TYPE_IPV6) {
+		// does this need both l3_src and l3_dst to be IPV6 or only l3_dst (important for a proposed solution to other problems)
 		if (dp_is_ip6_in_nat64_range(key->l3_dst.ip6))
 			key->vnf_type = DP_VNF_TYPE_NAT;
```

Comment:
This is kind of bad wording, since of course both src and dest are of the same type here. But this is mostly a callback to the proposed change to `struct flow_key`

Reply:
It needs both l3_src and and l3_dst to be type IPv6 and it needs dest ipv6 to be in a specific range.